datareservoirio/client.py

```diff
@@ -411,7 +420,21 @@ def get(
         else:
             df = pd.DataFrame(columns=("index", "values")).astype({"index": "int64"})
 
-        series = df.set_index("index").squeeze("columns").loc[start:end].copy(deep=True)
+        try:
+            series = (
+                df.set_index("index").squeeze("columns").loc[start:end].copy(deep=True)
+            )
+        except KeyError as e:
+            logging.warning(
+                "KeyError. The data will be sorted to attempt to resolve the issue. Please note that this operation may take some time."
```

Comment:
I'd just rephrase this a bit, we don't have to mention KeyError. Something like "The timeseries you requested is not properly ordered. The data will be sorted to attempt to resolve the issue. Please note that this operation may take some time."